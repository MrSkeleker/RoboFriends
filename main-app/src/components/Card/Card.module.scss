.card {
    cursor: pointer;
    width: 250px;
    box-sizing: border-box;

    img {
        width: 200px;
        height: auto;
    }

    h2 {
        height: 50px;
    }
}