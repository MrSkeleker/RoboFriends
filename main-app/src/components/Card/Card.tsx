import React from 'react';
import styles from './Card.module.scss';

interface CardProps {
    id: number;
    name: string;
    username: string;
    email: string;
}

function Card(props: CardProps){
    const {id, name, email} = props;
    return (
    <div className={`tc bg-light-green dib br4 pa3 ma2 grow bw2 shadow-5 ${styles.card}`}>
        <img src={`https://robohash.org/robot${id}?size=200x200`} alt='robot' />
        <div>
            <h2>{name}</h2>
            <p>{email}</p>
        </div>
    </div>)
}

export default Card;