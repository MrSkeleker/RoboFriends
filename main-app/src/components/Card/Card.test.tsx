import {shallow} from 'enzyme';
import React from 'react';
import Card from './Card';

const testProps = {
    id: -1,
    name: 'Test',
    email: 'test@test.com'
}

it('expect to render Card Component', () => {
    expect(shallow(<Card {...testProps}/>).length).toEqual(1);
    expect(shallow(<Card {...testProps}/>)).toMatchSnapshot();
});