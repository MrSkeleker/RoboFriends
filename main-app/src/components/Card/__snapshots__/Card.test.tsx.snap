// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card Component 1`] = `
<div
  className="tc bg-light-green dib br4 pa3 ma2 grow bw2 shadow-5 card"
>
  <img
    alt="robot"
    src="https://robohash.org/robot-1?size=200x200"
  />
  <div>
    <h2>
      Test
    </h2>
    <p>
      test@test.com
    </p>
  </div>
</div>
`;
